### ISSUE TITLE
failed to build passthrough filter metadata invalid type: jwt.MapClaims

### ISSUE THREAD

**Issue by edubonifs** [solo-io/gloo#9070]
### Gloo Edge Product

Enterprise

### Gloo Edge Version

1.16.0

### Kubernetes Version

1.25

### Describe the bug

One of our customers is having this error on 1.16.0 which wasn't present before:
```
{"level":"warn","ts":"2024-01-16T15:18:25Z","logger":"ext-auth.ext-auth-service","msg":"failed to build passthrough filter metadata from AuthorizationRequest.State:map[access_token:
...
]","version":"undefined","x-request-id":"7e8303a7-2609-4ece-b72b-451968d6cacb","error":"proto: invalid type: jwt.MapClaims"}
```

On the error log you they see that the OIDC flow finished correctly, and the ID token and access tokens are correctly fetched. But it fails to propagate them on the FilterMetadata field. What we see on the auth service is:
```
Could not extract id token or access token from gloo extauth-ee, metafields: map[io.solo.filters.http.sanitize:fields:{key:"custom_auth_server_name" value:{string_value:"solo.io.extauth.default_12345"}} solo.auth.passthrough: solo.auth.passthrough.config:]
```

This is their Authconfig, the pass through service does generate a jwt… but to do that it needs the access token or the id token in the context but this is failing before calling the pass through service:
```
apiVersion: enterprise.gloo.solo.io/v1
kind: AuthConfig
metadata:
  name: pv
  namespace: namespace
spec:
  configs:
  - passThroughAuth:
      config:
        redirectWorkaroundOnly: /oauth2/callback?
      grpc:
        address: .svc.cluster.local:4000
        connectionTimeout: 6s
  - oauth2:
      oidcAuthorizationCode:
        appUrl: https://dashboard/
        callbackPath: /oauth2/callback
        clientId: dashboard-pv
        clientSecretRef:
          name: gloo-oauth
          namespace: namespace
        headers:
          accessTokenHeader: access-token
        issuerUrl: https://oidc/
        scopes:
        - email
        - offline_access
        session:
          cookieOptions:
            maxAge: 86400
          redis:
            allowRefreshing: true
            cookieName: x-auth
            keyPrefix: id_token
            options:
              host: session-eu-west-1:6379
  - passThroughAuth:
      grpc:
        address: .svc.cluster.local:4000
        connectionTimeout: 6s
```

This is the related PR I have found out: https://github.com/solo-io/solo-projects/pull/5440

### Expected Behavior

It should work as it did on 1.15.10

### Steps to reproduce the bug

Attached Authconfig above

**Comment by nfuden** [comment]
Oauth2 sets the metadata here: https://github.com/solo-io/ext-auth-service/blob/master/pkg/config/oidc/oidc_request.go#L1043
Which we then try to marshal here: https://github.com/solo-io/ext-auth-service/blob/master/pkg/config/passthrough/grpc/grpc_request_adapter.go#L90

**Comment by bewebi** [comment]
I've identified solo-io/ext-auth-service#659 as the PR that introduced the bug
The `oidc_claims` added to the `state` have type `jwt.MapClaims` which is not a type that can be processed by passthrough

I am working on identifying the optimal solution

**Comment by bewebi** [comment]
The fix was merged to `master` in `ext-auth-service` but will still need to be:

- [ ] released in ext-auth-service v0.55
- [ ] brought in by solo-projects v1.17 beta via v0.55 and released
- [ ] brought in by solo-projects v1.16 via v0.55 and released

The bug is not present in versions of ext-auth earlier than v0.54 and therefore is not present in Gloo EE releases relying on earlier versions
Gloo EE 1.16 is the first to rely on extauth >= v0.54

**Comment by bewebi** [comment]
The fix has merged in `main` and `v1.16.x` and will be released in EE 1.17.0-beta1 and 1.16.1
